bleve-updates (#5246)
mattermost/docs: source/deploy/bleve-search.rst

The Bleve search engine works as a library integrated into the Mattermost codebase. As it generates indexes in the filesystem of the server that it is running on, it doesn’t require an external server to function. Because of this, Bleve should not be enabled in High Availability deployments.

.. note::

>   From Mattermost v6.1, Bleve search uses the scorch index type on newly created indexes which features efficiency improvements and indexes that use significantly less disk space. 

Reviewer: ```suggestion
  From Mattermost v6.1, Bleve search uses the scorch index type on newly-created indexes. This new index type features efficiency improvements and indexes that use significantly less disk space.
```

>   To use this new type of index, after upgrading Mattermost Server to version to v6.1 or later, go to **System Console > Experimental > Bleve**, then run a purge operation by selecting **Purge Index**, followed by a reindex by selecting **Index Now**. Bleve remains compatible with existing indexes, so currently indexed data will continue to work if a purge and reindex isn't run.

Reviewer: ```suggestion
  To use this new type of index on your existing indexes, first upgrade Mattermost Server to version to v6.1 or later. Then go to **System Console > Experimental > Bleve** and select **Purge Index** to run a purge operation. When that's complete, select **Index Now** to reindex. Bleve remains compatible with existing indexes, so currently indexed data will continue to work if a purge and reindex isn't run.
```

1. Open **System Console > Experimental > Bleve**.
2. Set **Enable Bleve Indexing** to **true** to enable the other settings on the page.

> 3. Set the directory path to use for storing bleve indexes (e.g.: ``/var/opt/mattermost/bleveindexes``). The user running Mattermost should have permissions to access the directory. See our `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html#bleve-settings>`__  documentation for details.

Reviewer: ```suggestion
3. Set the directory path to use for storing Bleve indexes (e.g.: ``/var/opt/mattermost/bleveindexes``). The user running Mattermost should have permissions to access the directory. See our `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html#bleve-settings>`__  documentation for details.
```
